### ISSUE TITLE
[API] SegmentInfo always returns an empty table.

### ISSUE THREAD

**Issue by HonorWOW** [Tercioo/Details-Damage-Meter#541]
**Describe the bug**
The Details API function [SegmentInfo](https://github.com/Tercioo/Details-Damage-Meter/blob/e2cc0b46c507526b871365049e520c9e83cc167f/functions/api2.lua#L120) will always return an empty table instead segment information.

Not a whole lot more to add -- we define segmentInfo as an empty table, and never give it a "real" value. Whether getCombatObject returns nil or not, we return the empty table unaltered.

I don't know the "correct" thing to return -- combatObject?

**Comment by Flamanis** [comment]
The function was implemented in a separate location and I guess that entry just was not updated. The function you'd want to use is `Details:GetSegmentInfo(segment)`

Returns are 
enemy, color, raid_type, killed, is_trash, portrait, background, background_coords

Enemy being either the boss name, arena name, or whatever npc got tagged first. 
color is boss only and dictates the content as well as a wipe/kill.
raid type being "party" or "raid"
killed being true/false for raid bosses
Is trash is only if the combat is not boss/arena and is counted as trash.
Portrait is the boss portrait image
background and background coords exist on boss and arena and are the background of the arena map, or the boss journal entry. (usually ir looks like)

**Comment by HonorWOW** [comment]
Fair enough If SegmentInfo isn't necessary anymore, it should probably be removed.

**Comment by Flamanis** [comment]
I don't disagree. API information needs an overhaul and I'll try to get to it during this week. It's a lot of work to get everything back to cleanliness.

**Comment by Flamanis** [comment]
Closing this issue as it is old and could have already been fixed. If you believe this to be in error, or if the issue still persists, please create a new issue.
